zxlive/common.py

```diff
@@ -1,22 +1,34 @@
+from __future__ import annotations
+
 import os
 from enum import IntEnum
-from typing import Final, Dict, Any, Optional
+from typing import Final, Optional, TypeVar, Type
 
 from pyzx import EdgeType
 from typing_extensions import TypeAlias
 
 from PySide6.QtCore import QSettings
-from PySide6.QtGui import QColor
-from PySide6.QtWidgets import QTabWidget
 
 import pyzx
 
 _ROOT = os.path.abspath(os.path.dirname(__file__))
 
 
+T = TypeVar('T')
+
+
 def get_data(path: str) -> str:
     return os.path.join(os.environ.get("_MEIPASS", _ROOT), path)
 
+
+def get_settings_value(arg: str, _type: Type[T], default: T | None = None, settings: QSettings | None = None) -> T:
```

Comment:
I was looking up what `Type[T]` meant as I was not familiar with it. I found that `typing.Type` is depricated as the builtin `type` now supports indexing. See https://docs.python.org/3/library/typing.html#typing.Type